modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -120,9 +120,10 @@ class StateChannel[
 
   private val runConsensus: Pipe[F, BlockConsensusInput, FinalBlock] =
     _.evalTap(input => logger.debug(s"Received block consensus input to process: ${input.show}"))
-      .evalMap(
-        new BlockConsensusCell[F](_, blockConsensusContext)
-          .run()
+      .evalMap(blockConsensusInput =>
+        OptionT(storages.lastSnapshot.getOrdinal)
+          .getOrRaise(new Exception("last snapshot ordinal required"))
```

Comment:
I see such exception in `BlockService` as well. I am wondering about the best approach to handle DRY.

Maybe introducing 2 methods: `getOrdinal: SnapshotOrdinal` which throws and `getOrdinalOption`?
Let me know what you think.

Reply:
I could make this change to the trait, if that's agreeable.
```
def getOrdinal(implicit mt: MonadThrow[F]): F[SnapshotOrdinal] =
  OptionT(getOrdinalOption)
    .getOrRaise(new Exception("last snapshot ordinal required"))
```
cc: @marcinwadon

Reply:
What trait you mention @sdavidp ?

Reply:
@marcinwadon I was referring to `LastSnapshotStorage` -- but the code has been refactored again and this part removed.

Reply:
@sdavidp I would change "last snapshot ordinal required" to match other errors, but also I'd explicitly say that this state is unexpected/forbidden. what about: "Unexpected state - could not find the latest snapshot ordinal"?

Reply:
@marcinwadon What about `IllegalStateException("could not find the latest snapshot ordinal")`

Reply:
fine for me, however, start with the uppercase as we do for other messages :D